components/automate-deployment/pkg/client/restore.go

```diff
@@ -187,60 +187,73 @@ func (r *DeploymentRestore) loadService() error {
 	return nil
 }
 
-func (r *DeploymentRestore) stopDeploymentService() error {
-	if err := r.target.UnloadService(r.service); err != nil {
+func (r *DeploymentRestore) stopDeploymentService(ctx context.Context) error {
+	ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
+	defer cancel()
+
+	if err := r.target.UnloadService(ctx, r.service); err != nil {
 		return status.Wrap(
 			err,
 			status.ServiceUnloadError,
 			"Stopping the deployment-service prior to restoration failed",
 		)
 	}
 
-	startTime := time.Now()
-	timeout := 10 * time.Second
-	sleepTime := 500 * time.Millisecond
-	ctx := context.Background()
+	ticker := time.NewTicker(500 * time.Millisecond)
+	defer ticker.Stop()
 
-	// Wait for Habitat to "unload" the process
-	for {
-		deployed, err := r.target.DeployedServices(ctx)
-		if err == nil {
-			alive := false
-			for _, svc := range deployed {
-				if svc.Pkg.Name() == deploymentServiceName {
-					alive = true
-				}
-			}
+	deploymentServiceRunning := func() (bool, error) {
+		var (
```

Comment:
does this all need to be forward declared?

Reply:
I'm was going to write a thesis on why forward declaring variables in the code is 10 million times better than relying on editors/IDE's/gopsl for type informtion because I don't have to context switch to see what things are, especially since the compiler doesn't care if you forward declare things, but instead I'll just leave this comment. :dancer:

Reply:
fair enough